zerocopy-derive/tests/ui-nightly/absence_of_deprecated_warning.rs

```diff
@@ -0,0 +1,24 @@
+//! See: https://github.com/google/zerocopy/issues/553
+//! zerocopy must still allow derives of deprecated types.
+//! This test has a hand-written impl of a deprecated type, and should result in a compilation
+//! error. If zerocopy does not tack an allow(deprecated) annotation onto its impls, then this
+//! test will fail because more than one compile error will be generated.
+#![deny(deprecated)]
+
+extern crate zerocopy;
+
+use zerocopy::IntoBytes;
+
+#[deprecated = "Do not use"]
+#[derive(IntoBytes)]
+#[repr(C)]
+struct OldHeader {
+    field_a: usize,
+    collection: [u8; 8],
+}
+
+trait T {}
+
+impl T for OldHeader {}
```

Comment:
What purpose does this serve?

Reply:
The useless impl triggers the specific compiler error we want. That is, zerocopy is implementing traits for the derived type as well, but only this impl should become an error.

Reply:
That's clever! Could you add a comment to that effect; e.g.:
```suggestion
trait T {}

// Intentionally trigger a deprecation error
impl T for OldHeader {}
```

Reply:
Sure!